Visualize application rewrite - OSD PR3 (#12324)
opensearch-project/documentation-website: _dashboards/visualize/multi-dimensional-data.md

   If your visualization displays a different value, make sure that your time filter window is large enough to encompass all the sample flight data.
   {: .note}

1. in the **Buckets** panel, select **Add**.

1. In the **Add bucket** drop-down, select **X-axis** to bucket the data by vertical column.

1. In the **Aggregation** drop-down, select an aggregation type for the rows.

   For this example, select `Histogram` to compare a series of values.

1. In the Field drop-down, select a field for the row buckets.

   For this example, select the numeric field `DistanceKilometers`.

1. Select **Update** in the lower right of the visualization tool to redraw the visualization.

   The chart shows the average ticket price as a function of bucketed flight distance.

1. in the **Buckets** panel, select **Add**.

1. In the **Add bucket** drop-down, select **Split series** to bucket the data by vertical column.

1. In the **Sub aggregation** drop-down, select an aggregation type for the rows.

   For this example, select `Terms` to compare different discrete values.

1. In the **Field** drop-down, select an aggregation type for the rows.

>    For this example, select `Day of week`.

Author: ```suggestion
   For this example, select `dayOfWeek`.
```

1. In the **Metrics** panel in the **Data** tab, select {::nomarkdown}<img src="{{site.url}}{{site.baseurl}}/images/icons/arrow-down-icon.png" class="inline-icon" alt="expand icon"/>{:/} (expand) **Metric Count** to expand the **Metrics** panel.

1. In the **Aggregation** box, choose the aggregation you want to display.

   For this tutorial, select `Average`.

1. In the **Field** box, choose the field you want to display.

   For this tutorial, select `FlightDelayMin`.

1. (Optional) To replace the raw field name in the visualization, enter an alternative name for the data field in the **Custom label** box.

   For this tutorial, enter `Flight delay in minutes`.

1. Select **Update** in the lower right of the visualization tool to view the metric in the display panel.

   The visualization panel displays a single bar with height of about `47`, the average flight delay for every document in the flight database.

   If your visualization displays a different value, make sure that your time filter window is large enough to encompass all the sample flight data.
   {: .note}

1. From the **Add bucket** drop-down, select **X-axis**.

> 1. In the **Add bucket** drop-down, select **X-axis** to bucket the data by vertical column.

Reviewer: Duplicate step — line 59 already says "From the **Add bucket** drop-down, select **X-axis**." This second one should be removed (or merged into the line above as the explanatory phrase "to bucket the data by vertical column").

Author: ```suggestion
```

   For this tutorial, there are only eight terms in the bucketing metric, so it's reasonable to display all of them. Do the following:

   1. In the **Order by** drop-down, select `Metric: FlightDelayMin`.

   1. In the **Size** box, enter or select the number of terms to display. For this example, choose `8`.

      Any number eight or higher displays all of the available `OriginWeather` terms.
      {: .note}

   1. In the **Order** drop-down, select `Descending`.

1. Select **Update** in the lower right of the visualization tool to redraw the visualization.

   As shown in the following image, the chart shows the average delay for each weather type at the flight origin. The greatest average delay was for `Damaging Wind`.

      <img src="{{site.url}}{{site.baseurl}}/images/dashboards/example-bar-chart-flight-delay.png" width="85%"  alt="Aggregation-based area chart"/>

1. Save the visualization as described in [Saving a data visualization]({{site.url}}{{site.baseurl}}/dashboards/visualize/saving-a-viz/).

## Building an area chart

> An area chart is a line chart with the area below the line shaded with a color. Multiple buckets can be stacked to show relative proportions of an running absolute value of the variable. Or, you can superimpose buckets or even different variables to compare within x-axis buckets or time values.

Reviewer 2: "an running" → "a running"?

Author: ```suggestion
An area chart is a line chart with the area below the line shaded with a color. Multiple buckets can be stacked to show relative proportions of a running absolute value of the variable. Or, you can superimpose buckets or even different variables to compare within x-axis buckets or time values.
```

In this tutorial, you'll use the OpenSearch sample flight data to create a visualization of the top five logs for flights delayed for every three hours over the last seven days.

To create the area chart, follow these steps:

1. Open a new visualization of type **Area**. See [Creating a new visualization]({{site.url}}{{site.baseurl}}/dashboards/visualize/new-viz/).

   For this tutorial, in the **New Area/Choose a source** dialog, choose `opensearch_dashboards_sample_data_flights`.

1. In the visualization tools, select the **Data** tab.

1. In the **Metrics** panel in the **Data** tab, select {::nomarkdown}<img src="{{site.url}}{{site.baseurl}}/images/icons/arrow-down-icon.png" class="inline-icon" alt="expand icon"/>{:/} (expand) **Y-axis Count** to expand the **Metrics** list.

1. In the **Aggregation** box, select the aggregation you want to display.

>    For this tutorial, select `Count`. Note that the `Count` metric there is no Field selection. This is because `Count` denotes the number of documents, not a field value.

Reviewer: Sentence is missing a word — "the `Count` metric **there is** no Field selection" doesn't parse. Suggested fix:

```suggestion
   For this tutorial, select `Count`. Note that for the `Count` metric there is no Field selection. This is because `Count` denotes the number of documents, not a field value.
```

   The Visualize application generates a vertical bar for each of the eight origin weather conditions.

1. From the Buckets panel, select **Add**.

1. From the **Add bucket** drop-down, select **Y-axis**.

1. From the **Sub aggregation** drop-down list, select **Terms**.

1. From the **Field** dropdown list, select **DestWeather**.

1. Select the same display options for the `DestWeather` term:

   - **Order by Metric**: `AverageFlightDelayMin`
   - **Order**: `Descending`
   - **Number**: `8`
   
1. Select **Update** to reflect these parameters in the graph.

1. Adjust the display of the heatmap by doing the following:

   1. Select the **Options** tab.

   1. In the **Heatmap settings** panel, select `Greys` from the **Color schema** drop-down.

   1. Enter `8` in the **Number of colors** box.

   1. In the **Labels** panel, select **Show labels**.

   1. Select *Update*.

   The visualization shows average flight delay mapped to origin and destination weather combinations as shown in the following image.

   <img src="{{site.url}}{{site.baseurl}}/images/dashboards/example-heatmap-flight-delay.png" width="85%"  alt="Aggregation-based area chart"/>

   Note the following:

>    - The heat map data is organized into four quadrants, corresponding to the four combinations of "good" (Cloudy, Rain, Clear, Sunny) and "bad" (Hail, Heavy Fog, Damaging Wind, Thunder & Lightening) weather conditions at the destination and origin. This is coincidental, but the interaction pattern is pronounced enough that it's visible if you order one or both buckets differently.

Reviewer: Typo: `Lightening` (the verb) → `Lightning` (the weather phenomenon). This must match the actual sample-data field value, which is `Thunder & Lightning` in `opensearch_dashboards_sample_data_flights`.

```suggestion
   - The heat map data is organized into four quadrants, corresponding to the four combinations of "good" (Cloudy, Rain, Clear, Sunny) and "bad" (Hail, Heavy Fog, Damaging Wind, Thunder & Lightning) weather conditions at the destination and origin. This is coincidental, but the interaction pattern is pronounced enough that it's visible if you order one or both buckets differently.
```

>    - The longest delays occurred when the weather was good at both the origin and destination. The shortest delays occurred occurred when the weather was bad at both locations.

Reviewer: Doubled word: `occurred occurred`.

```suggestion
   - The longest delays occurred when the weather was good at both the origin and destination. The shortest delays occurred when the weather was bad at both locations.
```

1. Now assume that you're interested in flight cancellations. You can limit the heat map display to only cancelled flights.

>    You can filter the data using the [filter tool]({{site.url}}{{site.baseurl}}/dashboards/discover/filter-tool), but if the visualization is in a dashboard this filters the data for other visualizations as well. The following procedure limits the data only for this current visualization.

Reviewer: **This is one of the dead links failing the `jekyll-build` job.** The `filter-tool` link is missing its trailing slash, which the link checker treats as 404.

```suggestion
   You can filter the data using the [filter tool]({{site.url}}{{site.baseurl}}/dashboards/discover/filter-tool/), but if the visualization is in a dashboard this filters the data for other visualizations as well. The following procedure limits the data only for this current visualization.
```

   For this tutorial, select `Average`.

1. In the **Field** box, choose the field you want to display.

   For this tutorial, select `AvgTicketPrice`.

1. (Optional) To replace the raw field name in the visualization, enter an alternative name for the data field in the **Custom label** box.

   For this tutorial, enter `Average Ticket Price`.

1. Select **Update** in the lower right of the visualization tool to view the metric in the display panel.

   The visualization panel displays a single bar with height of a little over `$600`, the average ticket price for every document in the flight database.

   If your visualization displays a different value, make sure that your time filter window is large enough to encompass all the sample flight data.
   {: .note}

1. in the **Buckets** panel, select **Add**.

1. In the **Add bucket** drop-down, select **X-axis** to bucket the data by vertical column.

1. In the **Aggregation** drop-down, select an aggregation type for the rows.

   For this example, select `Histogram` to compare a series of values.

1. In the Field drop-down, select a field for the row buckets.

>    For this example, select the text value `DistanceKilometers`.

Reviewer: `DistanceKilometers` is a numeric field, not a text field. The Histogram aggregation only applies to numeric fields, so the wording contradicts the procedure.

```suggestion
   For this example, select the numeric field `DistanceKilometers`.
```

1. Select **Update** in the lower right of the visualization tool to redraw the visualization.

   The chart shows the average ticket price as a function of bucketed flight distance.

1. in the **Buckets** panel, select **Add**.

> 1. In the **Add bucket** drop-down, select **Split series** to bucket the data by vertical column.

Reviewer: Two issues here:
1. **Wrong description** — `Split series` does *not* "bucket the data by vertical column"; that's the X-axis. Split series creates one line per term value, sharing the same X-axis.
2. There's also a duplicate `Add` step at line 328 (already opened the bucket panel) that probably needs trimming.

Suggested replacement for this single line:

```suggestion
1. In the **Add sub-buckets** drop-down, select **Split series** to create a separate line for each term.
```
